src/wallet/rpcwallet.cpp

```diff
@@ -2345,6 +2346,74 @@ Value listunspent(const Array& params, bool fHelp)
     return results;
 }
 
+Value zc_benchmark(const json_spirit::Array& params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp)) {
+        return Value::null;
+    }
+
+    if (fHelp || params.size() < 2) {
+        throw runtime_error(
+            "zcbenchmark benchmarktype samplecount\n"
+            "\n"
+            "Runs a benchmark of the selected type samplecount times,\n"
+            "returning the the running times of each sample.\n"
+            "\n"
+            "Output: [\n"
+            "  {\n"
+            "    \"runningtime\": runningtime\n"
+            "  },\n"
+            "  {\n"
+            "    \"runningtime\": runningtime\n"
+            "  }\n"
+            "  ...\n"
+            "]\n"
+            );
+    }
+
+    LOCK(cs_main);
+
+    std::string benchmarktype = params[0].get_str();
+    int samplecount = params[1].get_int();
+
+    if (samplecount <= 0) {
+        throw JSONRPCError(RPC_TYPE_ERROR, "Invalid samplecount");
+    }
+
+    double *sample_times = new double[samplecount];
+
+    if (benchmarktype == "createjoinsplit") {
+        // HACK: Do a JoinSplit to get the proving key loaded as a side-effect.
+        benchmark_create_joinsplit();
```

Comment:
Sean on slack: "make sure you're doing a fake joinsplit there -- like one that will immediately fail, instead of one that creates a valid proof"